crates/ruff_python_ast/src/nodes.rs

```diff
@@ -1351,6 +1351,64 @@ impl Ranged for FStringPart {
     }
 }
 
+pub trait StringFlags: Copy {
```

Comment:
Nice work! I had something similar in mind but I thought of using associated type for prefix:

```rs
pub trait StringFlags: Copy {
	type Prefix;

	fn prefix(self) -> Self::Prefix;
}
```